server/src/socket/index.js

```diff
@@ -0,0 +1,58 @@
+import mongoose from 'mongoose';
+
+import Users from '../Schemas/users';
+import Messages from '../Schemas/messages';
+import Channels from '../Schemas/channels';
+import { io } from '../setup';
+import {
+  newMessageEvent,
+  connectedUserAckEvent,
+  connectedUserEvent,
+  fetchOldChannelMessagesEvent,
+} from './eventNames';
+
+const connectedUserData = {};
+io.on('connection', async (client) => {
+  client.on(connectedUserEvent, async ({ userName }) => {
+    const user = await Users.findOne({ username: userName });
```

Comment:
Try to add as much tests as possible for the backend part, as js is a funky language. Also here you are managing state in a global variable `connectedUserData` which should be tested. You can make this storage persistence by storing in DB though, but for just setupping part it is ok

Reply:
@nimish-gupta @rollaball 
Added test cases to test socket events.
One test case is testing whether socket connections are being persisted to `connectedUserData`.

From code I felt that It will require refactor very soon, so not written test cases for all the events.
Please check the current coverage 
![test-socket-nuslack](https://user-images.githubusercontent.com/10244949/61000697-a0777900-a37b-11e9-946d-bf81d53b4d1b.png)